src/Common/src/TypeSystem/Common/TargetDetails.cs

```diff
@@ -186,5 +186,23 @@ public bool IsWindows
                 return OperatingSystem == TargetOS.Windows;
             }
         }
+
+        /// <summary>
+        /// Maximum number of elements in a HFA type.
+        /// </summary>
+        public int MaximumHfaElementCount
+        {
+            get
+            {
+                // There is a hard limit of 4 elements on an HFA type, see
+                // http://blogs.msdn.com/b/vcblog/archive/2013/07/12/introducing-vector-calling-convention.aspx
+                Debug.Assert(Architecture == TargetArchitecture.ARM ||
+                    Architecture == TargetArchitecture.ARM64 ||
+                    Architecture == TargetArchitecture.X64 ||
```

Comment:
The concept of an HFA is not just for parameter passing. The UTC compiler backend relies on the typesystem to identify opportunities to use the floating point vector registers within a function as well as at function boundaries. Arguably, we could build another abstraction here, but I don't see why we would. Perhaps we should use a name such as ValueTypeShapeCharacteristics so that we don't imply the HFA'ness is exclusive to the calling convention.